crates/bevy_render/src/renderer/render_device.rs

```diff
@@ -170,13 +169,13 @@ impl RenderDevice {
         &self.device
     }
 
-    pub fn map_buffer(&self, buffer: &wgpu::BufferSlice, map_mode: wgpu::MapMode) {
-        let data = buffer.map_async(map_mode);
-        self.poll(wgpu::Maintain::Wait);
-        assert!(
-            future::block_on(data).is_ok(),
-            "Failed to map buffer to host."
-        );
+    pub fn map_buffer(
+        &self,
+        buffer: &wgpu::BufferSlice,
+        map_mode: wgpu::MapMode,
+        callback: impl FnOnce(Result<(), BufferAsyncError>) + Send + 'static,
+    ) {
+        buffer.map_async(map_mode, callback);
```

Comment:
I'm confused here. This looks like you made what was previously synchronous (i.e. it would previously wait until the buffer had been mapped) to now being asynchronous (i.e. you now have to call poll or so somewhere else to trigger the callback) - why? I'm not sure how much this function is used though, and I'm not sure of the best way of using these async callbacks yet.

Reply:
the Changelog for this https://github.com/gfx-rs/wgpu/blob/master/CHANGELOG.md#map_async
specifically says 
> you aren't supposed to block and wait on the future to resolve

Without making a guess on how this method is used (it's not in Bevy), the best way to expose it is to pass the callback from the user